server/pulp/plugins/util/metadata_writer.py

```diff
@@ -1,47 +1,51 @@
-from gettext import gettext as _
 import glob
 import gzip
+import hashlib
 import logging
 import os
 import shutil
 import traceback
-
-
+from gettext import gettext as _
+from xml.dom import minidom
 from xml.sax.saxutils import XMLGenerator
 
 from pulp.common import error_codes
-from pulp.server.exceptions import PulpCodedValidationException, PulpCodedException
-from verification import CHECKSUM_FUNCTIONS
+from pulp.server.exceptions import PulpCodedException, PulpCodedValidationException
 
 _LOG = logging.getLogger(__name__)
 BUFFER_SIZE = 1024
+HASHLIB_ALGORITHMS = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
 
 
 class MetadataFileContext(object):
     """
     Context manager class for metadata file generation.
     """
 
-    def __init__(self, metadata_file_path, checksum_type=None):
+    def __init__(self, metadata_file_path, checksum_type=None, no_checksum_filenames=None):
         """
         :param metadata_file_path: full path to metadata file to be generated
         :type  metadata_file_path: str
         :param checksum_type: checksum type to be used to generate and prepend checksum
-                              to the file names of files. If checksum_type is None,
+                              to the file names of metadata files. If checksum_type is None,
                               no checksum is added to the filename
         :type checksum_type: str or None
+        :param no_checksum_filenames: file names that *will not* have a checksum prepended to them
+        :type no_checksum_filenames: list or None
```

Comment:
Nitpick, but maybe name this "dont_checksum_filenames"? It's longer which is a negative, but it's more verby and maybe clearer (not that the param docs aren't clear). Feel free to ignore.